framework/src/postprocessors/FindValueOnLine.C

```diff
@@ -62,35 +62,48 @@ FindValueOnLine::execute()
   Real s_right = 1.0;
   Real right = getValueAtPoint(_end_point);
 
-  for (unsigned i = 0; i < _depth; ++i)
+  /**
+   * Here we determine the direction of the solution. i.e. the left might be the high value
+   * while the right might be the low value.
+   */
+  bool left_to_right = left < right;
+  // Initial bounds check
+  if ((left_to_right && _target < left) || (!left_to_right && _target < right))
+    mooseError("Target value \"" << _target << "\" is less than the minimum sampled value \"" << std::min(left, right) << "\"");
+  if ((left_to_right && _target > right) || (!left_to_right && _target > left))
+    mooseError("Target value \"" << _target << "\" is greater than the maximum sampled value \"" << std::max(left, right) << "\"");
+
+  bool found_it = false;
+  Real value = 0;
+  for (auto i = decltype(_depth)(0); i < _depth; ++i)
   {
     // find midpoint
     s = (s_left + s_right) / 2.0;
     Point p = s * (_end_point - _start_point) + _start_point;
 
     // sample value
-    Real value = getValueAtPoint(p);
+    value = getValueAtPoint(p);
 
     // have we hit the target value yet?
     if (MooseUtils::absoluteFuzzyEqual(value, _target, _tol))
+    {
+      found_it = true;
       break;
+    }
 
     // bisect
-    if (   (left >= _target && _target >= value)
-        || (left <= _target && _target <= value))
-    {
+    if ((left_to_right && _target < value) ||
+        (!left_to_right && _target > value))
       // to the left
       s_right = s;
-      right = value;
-    }
     else
-    {
       // to the right
       s_left = s;
-      left = value;
-    }
   }
 
+  if (!found_it)
+    mooseError("Target value \"" << std::setprecision(10) << _target << "\" not found on line within tolerance, last sample: " << value << ".");
```

Comment:
Me neither, I just put this in because it's the reason your tolerance isn't being honored right now. This is your design, you tell me what you'd like.

We probably need to do something because if the underlying assumption of a monotonic function is violated then we may be really far off when we get down here and that's after the new initial bounds checks were added.